File: spaghetti-typescript-support/src/integTest/module/src/TestModule.ts
import * as spaghetti_test_dependency from "spaghetti_test_dependency";
import { Numbers, Point3d } from "TestModule.module";

declare var libWithVersion: any;

export class TestModule {
	static addTwoNumbers(a:number, b:number):number {
		return a + b;
	}

	static getNextEnumValue(value:spaghetti_test_dependency.Fruit):spaghetti_test_dependency.Fruit {
		return value + 1;
	}

	static getPositionInPrimes(value:spaghetti_test_dependency.Prime):string {
		return spaghetti_test_dependency.Prime[value];
	}

	static getValueOfTwo():string {
		return Numbers.TWO;
	}

	static getValueOfDependentConstant():string {
		return spaghetti_test_dependency.DependentConstant.A;
	}

	static createPoint3dWithGivenValues(x:number, y:number, z:number):Point3d {
		return {
			x: x,
			y: y,
			z: z
		};
	}

	static getPointFromDependencyModule():spaghetti_test_dependency.Point2d {
		return spaghetti_test_dependency.DependencyModule.getPoint();
	}

	static getPointFromDependencyModuleViaCallback(x:number, y:number):spaghetti_test_dependency.Point2d {
		var returnedPoint:spaghetti_test_dependency.Point2d = null;
		spaghetti_test_dependency.DependencyModule.createPointViaCallback(x, y, (point:spaghetti_test_dependency.Point2d) => {
			returnedPoint = point;
		});
		return returnedPoint;
	}

	static returnPointViaCallback(x:number, y:number, z:number, callback:(point:Point3d)=>void) {
		callback({x: x, y: y, z: z});
	}

    static getExternalDependencyVersion(): string {
		return libWithVersion.version;
    }
}

File: spaghetti-kotlin-support/src/integTest/module/src/spaghetti/test/TestModule.kt
package spaghetti.test

import spaghetti.test.dependency.*

public class TestModule {
	companion object {
		public fun addTwoNumbers(a:Int, b:Int): Int {
			return a + b
		}

		public fun getNextEnumValue(value:Fruit):Fruit {
			return ((value as Int) + 1) as Fruit
		}

		public fun getPositionInPrimes(value:Prime):String {
        	return Prime.getName(value)
        }

		public fun getValueOfTwo():String {
			return Numbers.TWO
		}

		public fun createPoint3dWithGivenValues(x:Int, y:Int, z:Int):Point3d {
			return object : Point3d {
				override val x = x
				override val y = y
				override val z = z
			}
		}

		public fun getPointFromDependencyModule():Point2d {
			return DependencyModule.getPoint()
		}

		public fun getPointFromDependencyModuleViaCallback(x:Int, y:Int):Point2d {
            var result:Point2d? = null
			DependencyModule.createPointViaCallback(x, y, { point ->
				result = point
			})
            return result!!
		}

		public fun returnPointViaCallback(x:Int, y:Int, z:Int, callback:(Point3d)->Unit) {
            callback(object : Point3d {
                override val x = x
                override val y = y
                override val z = z
            });
		}

		public fun getExternalDependencyVersion():String {
		    return js("libWithVersion.version")
		}
	}
}
